src/js/components/AppPageComponent.jsx

```diff
@@ -52,74 +56,196 @@ var AppPageComponent = React.createClass({
     return {
       activeViewIndex: 0,
       activeTabId: activeTabId,
-      tabs: tabs
+      activeTaskId: null,
+      app: AppsStore.getCurrentApp(appId),
+      tabs: tabs,
+      fetchState: States.STATE_LOADING
     };
   },
 
+  componentWillMount: function () {
+    AppsStore.on(AppsEvents.CHANGE, this.onAppChange);
+    AppsStore.on(AppsEvents.REQUEST_APP_ERROR, this.onAppRequestError);
+    AppsStore.on(AppsEvents.SCALE_APP_ERROR, this.onScaleAppError);
+    AppsStore.on(AppsEvents.RESTART_APP_ERROR, this.onRestartAppError);
+    AppsStore.on(AppsEvents.DELETE_APP_ERROR, this.onDeleteAppError);
+    AppsStore.on(AppsEvents.DELETE_APP, this.onDeleteAppSuccess);
+  },
+
+  componentWillUnmount: function () {
+    AppsStore.removeListener(AppsEvents.CHANGE,
+      this.onAppChange);
+    AppsStore.removeListener(AppsEvents.REQUEST_APP_ERROR,
+      this.onAppRequestError);
+    AppsStore.removeListener(AppsEvents.SCALE_APP_ERROR,
+      this.onScaleAppError);
+    AppsStore.removeListener(AppsEvents.RESTART_APP_ERROR,
+      this.onRestartAppError);
+    AppsStore.removeListener(AppsEvents.DELETE_APP_ERROR,
+      this.onDeleteAppError);
+    AppsStore.removeListener(AppsEvents.DELETE_APP,
+      this.onDeleteAppSuccess);
+  },
+
```

Comment:
What if we had a map of event -> handler. Like so, it could be in a function or as a prop:
```
this.eventMap = {
AppsEvents.CHANGE: this.onAppChange,
AppsEvents.REQUEST_APP_ERROR: this.onAppRequestError,
...
}
```

Then, in each one of these methods we do something like this:
```
componentWillMount: function () {
  this.getEventMap.forEach(function(listener, event) {
    AppStore.on(event, listener);
  });
},
```
then do the same for `componentWillUnmount`.